flink-core/src/main/java/org/apache/flink/configuration/GlobalConfiguration.java

```diff
@@ -241,6 +265,95 @@ private static Configuration loadYAMLResource(File file) {
         return config;
     }
 
+    /**
+     * Flattens a nested configuration map to be only one level deep.
+     *
+     * <p>Nested keys are concatinated using the {@code KEY_SEPARATOR} character. So that:
+     *
+     * <pre>
+     * keyA:
+     *   keyB:
+     *     keyC: "hello"
+     *     keyD: "world"
+     * </pre>
+     *
+     * <p>becomes:
+     *
+     * <pre>
+     * keyA.keyB.keyC: "hello"
+     * keyA.keyB.keyD: "world"
+     * </pre>
+     *
+     * @param config an arbitrarily nested config map
+     * @param keyPrefix The string to prefix the keys in the current config level
+     * @return A flattened, 1 level deep map
+     */
+    @SuppressWarnings("unchecked")
+    private static Map<String, Object> flatten(
+            @Nonnull Map<String, Object> config, String keyPrefix) {
```

Comment:
Flink parameters are by default non-null. Only nullable ones should be explicitly annotated with `@Nullable`.